go/ct/common/cpp_coverage.go

```diff
@@ -30,6 +30,6 @@ func isCppCoverageEnabled() bool {
 // Not calling this function will result in no coverage data being reported
 // for runtime loaded C and C++ libraries.
 // If coverage data collection is not enabled, this function is a no-op.
-func DumpCppCoverageData() {
+func DumpCppCoverageData() { // coverage-ignore nothing to test
```

Comment:
as I understood it, the only way to make a test for this function would be to start it with the cpp coverage flag enabled. And this is not something we will do when running UT/coverage.

Reply:
A simple call to this function making sure it does not panic of crash the test would be sufficient to get the coverage done.

The explanation "nothing to test" is a not acceptable, since "nothing to test" means also that this function has no purpose and should not exist.

As far as I remember, when @LuisPH3 added this function there have already been tests that make sure that if compiled with a special flag, dump data is actually produced. So this does not have to be tested again.

Reply:
I think this could be solved in a more elegant way by changing the order of the unit test.
```
       // write coverage data into tempDir directory
	tempDir := t.TempDir()
	os.Setenv("GCOV_PREFIX", tempDir)

	// run dump routine
	DumpCppCoverageData()
	
        expectEnabled := *stateImpl
	enabled := isCppCoverageEnabled()

	if !enabled {
		if expectEnabled {
			t.Fatalf("Failed, cpp coverage is not enabled")
		} else {
			t.Skip("Skipping test, cpp coverage disabled and not expected to be enabled")
		}
	} else if !expectEnabled {
		t.Fatalf("Failed, cpp coverage is enabled, but it was expected disabled")
	}
```

With this change, the function is always called, even when side-effect is to be ignored.